### ISSUE TITLE
changes to numa_node_size breaks program

### ISSUE THREAD

**Issue by DE0CH** [numactl/numactl#135]
In commit [a7c4bc7](https://github.com/numactl/numactl/commit/a7c4bc790a191d3e42b63850b409c1a72b75a4e1) the signature of the function is changed from

```
long numa_node_size(int node, long *freep)
```

to

```
long long numa_node_size(int node, long long *freep);
```

causing any program that uses this this function to fail to compile. 

While I see that this is done to prevent overflow, it shouldn't be introduced in a maintenance release. Since it is backward incompatible, it should be included in the next major release according to [Semantic Versioning](https://semver.org/).

**Comment by andikleen** [comment]
On Fri, Sep 16, 2022 at 05:50:18PM +0000, Deyao Chen wrote:
> In commit a7c4bc7 the signature of the function is changed from
> 
> long numa_node_size(int node, long *freep)
> 
> to
> 
> long long numa_node_size(int node, long long *freep);
> 
> causing any program that uses this this function to fail to compile.
> 
> While I see that this is done to prevent overflow, it shouldn't be introduced
> in a maintenance release. Since it is backward incompatible, it should be
> included in the next major release according to Semantic Versioning.

Agreed. I will revert it.

**Comment by thesamesam** [comment]
Using abidiff may help to find problems like this.

**Comment by filbranden** [comment]
Resolved with release of 2.0.16

**Comment by thesamesam** [comment]
> Using abidiff may help to find problems like this.

I should've elaborated a bit, here's some examples of how projects do it in CI: https://github.com/Matroska-Org/libebml/issues/104#issuecomment-1266794833. Cheers!
